### ISSUE TITLE
索引超出了数组界限。

### ISSUE THREAD

**Issue by xiadongxiang** [t1m0thyj/WinDynamicDesktop#175]
[2019-10-24 10:18:45.192] System.IndexOutOfRangeException: 索引超出了数组界限。
   在 WinDynamicDesktop.ThemeDialog..ctor()
   在 WinDynamicDesktop.ThemeManager.SelectTheme()
   在 WinDynamicDesktop.LaunchSequence.NextStep(Boolean themeReadyOverride)
   在 WinDynamicDesktop.LocationManager.OnLocationDialogClosed(Object sender, EventArgs e)
   在 System.Windows.Forms.Form.OnFormClosed(FormClosedEventArgs e)
   在 System.Windows.Forms.Form.WmClose(Message& m)
   在 System.Windows.Forms.Form.WndProc(Message& m)
   在 System.Windows.Forms.Control.ControlNativeWindow.OnMessage(Message& m)
   在 System.Windows.Forms.Control.ControlNativeWindow.WndProc(Message& m)
   在 System.Windows.Forms.NativeWindow.Callback(IntPtr hWnd, Int32 msg, IntPtr wparam, IntPtr lparam)

**Comment by dbshao** [comment]
我也有同样的问题

**Comment by t1m0thyj** [comment]
Translation to English:
> The index is outside the bounds of the array.

> I have the same problem

**Comment by raymonraymon** [comment]
How to tackle ?

**Comment by YuFulu** [comment]
@t1m0thyj How to tackle ?

**Comment by Jmmes42** [comment]
I have the same  problem

**Comment by YuFulu** [comment]
debug.log  
```  
./settings.conf
{
  "location": "redacted",
  "latitude": "0",
  "longitude": "0",
  "darkMode": false,
  "hideTrayIcon": false,
  "disableAutoUpdate": false,
  "lastUpdateCheck": null,
  "themeName": null,
  "useWindowsLocation": false,
  "language": "zh-Hans",
  "poeditorApiToken": null,
  "enableShuffle": false,
  "lastShuffleDate": null,
  "shuffleHistory": null,
  "dontUseLocation": true,
  "sunriseTime": "6:00:00",
  "sunsetTime": "18:00:00",
  "sunriseSunsetDuration": 120,
  "fullScreenPause": false,
  "enableScripts": false
}
```  
WinDynamicDesktop.exe.log  
```  
[2020-04-21 02:06:20.264] System.IndexOutOfRangeException: 索引超出了数组界限。
   在 WinDynamicDesktop.ThemeDialog..ctor()
   在 WinDynamicDesktop.ThemeManager.SelectTheme()
   在 WinDynamicDesktop.MainMenu.OnThemeItemClick(Object sender, EventArgs e)
   在 System.Windows.Forms.ToolStripItem.RaiseEvent(Object key, EventArgs e)
   在 System.Windows.Forms.ToolStripMenuItem.OnClick(EventArgs e)
   在 System.Windows.Forms.ToolStripItem.HandleClick(EventArgs e)
   在 System.Windows.Forms.ToolStripItem.HandleMouseUp(MouseEventArgs e)
   在 System.Windows.Forms.ToolStripItem.FireEventInteractive(EventArgs e, ToolStripItemEventType met)
   在 System.Windows.Forms.ToolStripItem.FireEvent(EventArgs e, ToolStripItemEventType met)
   在 System.Windows.Forms.ToolStrip.OnMouseUp(MouseEventArgs mea)
   在 System.Windows.Forms.ToolStripDropDown.OnMouseUp(MouseEventArgs mea)
   在 System.Windows.Forms.Control.WmMouseUp(Message& m, MouseButtons button, Int32 clicks)
   在 System.Windows.Forms.Control.WndProc(Message& m)
   在 System.Windows.Forms.ScrollableControl.WndProc(Message& m)
   在 System.Windows.Forms.ToolStrip.WndProc(Message& m)
   在 System.Windows.Forms.ToolStripDropDown.WndProc(Message& m)
   在 System.Windows.Forms.Control.ControlNativeWindow.OnMessage(Message& m)
   在 System.Windows.Forms.Control.ControlNativeWindow.WndProc(Message& m)
   在 System.Windows.Forms.NativeWindow.Callback(IntPtr hWnd, Int32 msg, IntPtr wparam, IntPtr lparam)  
```

**Comment by t1m0thyj** [comment]
This should be fixed in version 4.1.0 which was just released. The update may take 24 hours to show up on Microsoft Store.

**Comment by LeQuadratico** [comment]
Still have that problem on 4.1.1 #214
